pkg/auth/providers/migration/activedirectory_migration.go

```diff
@@ -96,6 +100,11 @@ func (m *adMigration) migrate() error {
 			if strings.HasPrefix(principal, "activedirectory_user://") && strings.Contains(principal, ",") {
 				err = m.migrateUser(user, principal)
```

Comment:
On question on "migrateUser" - If this method fails and returns any error, does the migration halt and not go through rest of the users? When the backoff retry kicks off, if it fails again at the same user, we might end up not trying migration of remaining users at all? Wondering if we should go through the entire list of users and attempt migration before retrying for the failed users.

Reply:
I did add the exception for "user no longer exists" in AD.  In that case, we just continue.  I suppose we could do it that way for all failure cases.

Reply:
So in the retry, when we try to go over list of users, hoping we skip the ones already migrated.

Reply:
Thinking about this more.  I definitely understand your point, but I feel like (after all of these fixes are in) the most likely scenario for a failure is an intermittent blip with AD, which should be resolved with retries, but if it isn't quickly resolved, we're probably better off going into our backoff loop.

Reply:
Ok, yeah if majority failures seen so far are due to AD connectivity, it better to fail-fast on the first user itself rather than attempting migration for other users and seeing the same error in logs over and over.